e2e/app.po.ts

```diff
@@ -130,24 +130,27 @@ export class AppPage {
     // need to disable angular wait before check for current url because we're being redirected outside of angular
     browser.waitForAngularEnabled(false);
 
-    this.goToUrl(AppPage.baseurl);
-
-    // wait either for login page or loaded ipaas app
-    await P.race([
-      browser.wait(ExpectedConditions.presenceOf(this.rootElement), 1000, 'ipaas root element - assuming we are already logged in'),
-      browser.wait(ExpectedConditions.presenceOf(element(by.css('input#login_field'))), 1000, 'Github login page' ).then(() => {
-        log.info('GitHub login page');
-        return new GithubLogin().login(user);
-      }),
-    ]);
-
-    await P.race([
-      browser.wait(ExpectedConditions.presenceOf(this.rootElement), 1000, 'ipaas root element - assuming we are already logged in'),
-      browser.wait(ExpectedConditions.presenceOf(element(by.css('input#firstName'))), 1000, 'Keycloack login page' ).then(() => {
-        log.info('Keycloak first time visit page');
-        return new KeycloakDetails().submitUserDetails(user.userDetails);
-      }),
-    ]);
+    await this.goToUrl(AppPage.baseurl);
+
+    let currentUrl = await browser.getCurrentUrl();
+    if (contains(currentUrl, 'github.com/login')) {
+      log.info('GitHub login page');
```

Comment:
Wonder if there is a better way to document and log this, possibly using the testing framework.